src/jrd/exe.cpp

```diff
@@ -1200,7 +1200,19 @@ void EXE_execute_triggers(thread_db* tdbb,
 					&tdbb->getAttachment()->att_original_timezone,
 					tdbb->getAttachment()->att_current_timezone);
 
-				EXE_start(tdbb, trigger, transaction);
+				if (trigger_action == TRIGGER_DISCONNECT)
+				{
+					if (!trigger->req_timer)
+						trigger->req_timer = FB_NEW_POOL(*tdbb->getAttachment()->att_pool) TimeoutTimer();
+
+					const unsigned int timeOut = tdbb->getDatabase()->dbb_config->getOnDisconnectTrigTimeout() * 1000;
+					trigger->req_timer->setup(timeOut, isc_cfg_stmt_timeout);
+					trigger->req_timer->start();
+					thread_db::TimerGuard timerGuard(tdbb, trigger->req_timer, true);
+					EXE_start(tdbb, trigger, transaction);
```

Comment:
A comment here would be good, like `// under timerGuard scope` as the line is identical to the "else" case and some wrong refactor may put both outside of the "if".